source/state_representation/include/state_representation/parameters/ParameterType.hpp

```diff
@@ -19,4 +22,72 @@ enum class ParameterType {
   VECTOR,
   MATRIX
 };
-}
\ No newline at end of file
+
+static const std::map<std::string, ParameterType> parameter_type_map = {
+    {"Bool", ParameterType::BOOL},     {"BoolArray", ParameterType::BOOL_ARRAY},
+    {"Int", ParameterType::INT},       {"IntArray", ParameterType::INT_ARRAY},
+    {"Double", ParameterType::DOUBLE}, {"DoubleArray", ParameterType::DOUBLE_ARRAY},
+    {"String", ParameterType::STRING}, {"StringArray", ParameterType::STRING_ARRAY},
+    {"State", ParameterType::STATE},   {"Vector", ParameterType::VECTOR},
+    {"Matrix", ParameterType::MATRIX}};
+
+/**
+ * @brief Return the parameter type map, containing mappings of human-friendly strings for the type to their corresponding enum type
+ * @return std::map<std::string, ParameterType> with label, ParameterType pairs
+ */
+[[maybe_unused]] static const std::map<std::string, ParameterType>& get_parameter_type_map() {
+  return parameter_type_map;
+}
+
+/**
+ * @brief Return the ParameterType that corresponds to a name
+ * @param name The name for lookup
+ * @throws std:out_of_range Exception if the name does not correspond to a ParameterType
+ * @return ParameterType that corresponds to the (std::string) name
+ */
+[[maybe_unused]] static ParameterType parameter_type_from_name(const std::string& name) {
+  ParameterType p;
+  try {
+    p = parameter_type_map.at(name);
+  } catch (...) {
+    throw std::out_of_range("This ParameterType does not exist.");
```

Comment:
nitpick: you could just as well format and include `name` in the error message for more clarity

Reply:
ah, yes, good thinking, I will add it.